src/vellum/workflows/context.py

```diff
@@ -0,0 +1,82 @@
+from contextlib import contextmanager
+from functools import wraps
+import inspect
+import threading
+from typing import Any, Callable, Dict, Iterator, TypeVar, Union, cast
+
+from vellum.workflows.events.types import ParentContext
+
+LoggingContext = Dict[str, Any]
+_CONTEXT_KEY = "_execution_context"
+
+local = threading.local()
+
+F = TypeVar('F', bound=Callable[..., Any])
+T = TypeVar('T')
+P = TypeVar('P', bound=ParentContext)
+
+
+def get_execution_context() -> LoggingContext:
+    """Retrieve the current execution context."""
+    return getattr(local, _CONTEXT_KEY, {})
+
+
+def set_execution_context(context: LoggingContext) -> None:
```

Comment:
If we don't intend for this to be used outside of this module (which I think is a good idea if we can get away with it) let's rename it to `_set_execution_context`